File: go_files/ex8.go
package main

import (
	"fmt"
)

type Empregado struct {
	nome    string
	cargo   string
	salario float64
}

func (e Empregado) Informar() string {
	return fmt.Sprintf("Nome: %s, Cargo: %s, Salário: %.2f", e.nome, e.cargo, e.salario)
}

type Empresa struct {
	nome       string
	empregados []*Empregado
}

func (emp *Empresa) AdicionarEmpregado(e *Empregado) {
	emp.empregados = append(emp.empregados, e)
}

func (emp Empresa) Informar() {
	fmt.Println("Empresa:", emp.nome)
	fmt.Println("Empregados:")
	for _, empregado := range emp.empregados {
		fmt.Println(empregado.Informar())
	}
}

func main() {
	empresa := Empresa{nome: "Softplan"}

	emp1 := &Empregado{nome: "Guilherme", cargo: "Programador", salario: 1000.00}
	emp2 := &Empregado{nome: "carlos", cargo: "Gerente", salario: 7000.00}
	emp3 := &Empregado{nome: "Joaquim", cargo: "Analista", salario: 9000.00}

	empresa.AdicionarEmpregado(emp1)
	empresa.AdicionarEmpregado(emp2)
	empresa.AdicionarEmpregado(emp3)

	empresa.Informar()
}

File: pyton_files/ex8.py


class Empregado:
    def __init__(self, nome, cargo, salario):
        self.nome = nome
        self.cargo = cargo
        self.salario = salario

    def exibir_detalhes(self):
        return f"Nome: {self.nome}, Cargo: {self.cargo}, Salário: R${self.salario:.2f}"

class Empresa:
    def __init__(self, nome):
        self.nome = nome
        self.empregados = []  

    def adicionar_empregado(self, empregado):
        self.empregados.append(empregado)

    def exibir_empregados(self):
        for empregado in self.empregados:
            print(empregado.exibir_detalhes())


def main():
    empresa = Empresa("Tech Solutions")

    
    empregado1 = Empregado("Carlos Silva", "Desenvolvedor", 5000)
    empregado2 = Empregado("Ana Souza", "Gerente de Projetos", 8000)
    empregado3 = Empregado("João Oliveira", "Analista de Sistemas", 6000)

    
    empresa.adicionar_empregado(empregado1)
    empresa.adicionar_empregado(empregado2)
    empresa.adicionar_empregado(empregado3)

    
    print(f"Empregados da empresa {empresa.nome}:")
    empresa.exibir_empregados()


if __name__ == "__main__":
    main()
